src/cmake/modules/FindLLVM.cmake

```diff
@@ -0,0 +1,111 @@
+#  LLVM_LIB_DIR     - where to find llvm libs
+#
+# The following input symbols may be used to help guide the search:
+#  LLVM_DIRECTORY   - the root of the LLVM installation (if custom)
+#  LLVM_FIND_QUIETLY - if true, will suppress most console output
+#  LLVM_STATIC      - if true, will prefer static LLVM libs to dynamic
+
+# try to find llvm-config, with a specific version if specified
+if (LLVM_DIRECTORY)
+    set (LLVM_CONFIG_PATH_HINTS "${LLVM_DIRECTORY}/bin")
+endif ()
+list (APPEND LLVM_CONFIG_PATH_HINTS
+        "/usr/local/opt/llvm/${LLVM_VERSION}/bin/"
+        "/usr/local/opt/llvm/bin/")
+find_program (LLVM_CONFIG
+              NAMES llvm-config-${LLVM_VERSION} llvm-config
+              HINTS ${LLVM_CONFIG_PATH_HINTS} NO_DEFAULT_PATH)
+find_program (LLVM_CONFIG
+              NAMES llvm-config-${LLVM_VERSION} llvm-config
+              HINTS ${LLVM_CONFIG_PATH_HINTS})
+if (NOT LLVM_FIND_QUIETLY)
+    message (STATUS "Found llvm-config '${LLVM_CONFIG}'")
+endif ()
+
+if (NOT LLVM_DIRECTORY)
+    execute_process (COMMAND ${LLVM_CONFIG} --prefix
+           OUTPUT_VARIABLE LLVM_DIRECTORY
+           OUTPUT_STRIP_TRAILING_WHITESPACE)
+endif()
+
+execute_process (COMMAND ${LLVM_CONFIG} --version
+       OUTPUT_VARIABLE LLVM_VERSION
+       OUTPUT_STRIP_TRAILING_WHITESPACE)
+execute_process (COMMAND ${LLVM_CONFIG} --libdir
+       OUTPUT_VARIABLE LLVM_LIB_DIR
+       OUTPUT_STRIP_TRAILING_WHITESPACE)
+execute_process (COMMAND ${LLVM_CONFIG} --includedir
+       OUTPUT_VARIABLE LLVM_INCLUDES
+       OUTPUT_STRIP_TRAILING_WHITESPACE)
+if (NOT ${LLVM_VERSION} VERSION_LESS 3.8)
+    execute_process (COMMAND ${LLVM_CONFIG} --system-libs
+                     OUTPUT_VARIABLE LLVM_SYSTEM_LIBRARIES
+                     OUTPUT_STRIP_TRAILING_WHITESPACE)
+else ()
+    # Older LLVM did not have llvm-config --system-libs, but we know that
+    # on Linux, we'll need curses.
+    find_package (Curses)
+    if (CURSES_FOUND)
+        list (APPEND LLVM_SYSTEM_LIBRARIES ${CURSES_LIBRARIES})
+    endif ()
+endif ()
+
+find_library ( LLVM_LIBRARY
+               NAMES LLVM-${LLVM_VERSION} LLVM
+               PATHS ${LLVM_LIB_DIR})
+find_library ( LLVM_MCJIT_LIBRARY
+               NAMES LLVMMCJIT
+               PATHS ${LLVM_LIB_DIR})
+
+# if (NOT LLVM_LIBRARY)
+#     execute_process (COMMAND ${LLVM_CONFIG} --libfiles engine
+#                      OUTPUT_VARIABLE LLVM_LIBRARIES
+#                      OUTPUT_STRIP_TRAILING_WHITESPACE)
+# endif ()
+
+# shared llvm library may not be available, this is not an error if we use LLVM_STATIC.
+if ((LLVM_LIBRARY OR LLVM_LIBRARIES OR LLVM_STATIC) AND LLVM_INCLUDES AND LLVM_DIRECTORY AND LLVM_LIB_DIR)
+  if (LLVM_STATIC)
+    # if static LLVM libraries were requested, use llvm-config to generate
+    # the list of what libraries we need, and substitute that in the right
+    # way for LLVM_LIBRARY.
+    execute_process (COMMAND ${LLVM_CONFIG} --libfiles
```

Comment:
Surprised this has been like this for so long.

To get the list of static libs we need `llvm-config --libfiles --link-static`

Reply:
Thanks for the tip. Submitted as https://github.com/imageworks/OpenShadingLanguage/pull/1070